### PR TITLE
Add support standalone es datastore for multi entity text detection 

### PR DESCRIPTION
**Pull request by ankur09011**
## JIRA Ticket Number

JIRA TICKET: 1891, 1890

## Description of change

Why:
* To add support for elastic search datastore
* To add support for multi entity text detection
* To add new endpoint for multi_entity_text detection

How:
* By creating package for es_datastore
* By adding function for generating and parsing es query and results
* By adding entry in urls and creating a placeholder view

## Checklist (OPTIONAL):

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: es_datastore/elastic_search.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+        self._check_doc_type_for_elasticsearch()
+        request_timeout = self._connection_settings.get('request_timeout', 20)
+        index_name = self._index_name
+
+        doc_type = self._connection_settings[
+            constants.ELASTICSEARCH_DOC_TYPE]
+
+        index_header = json.dumps({'index': self._index_name, 'type': doc_type})
+
+        data = list \
```
Comment (haptik-deployment) [inline es_datastore/elastic_search.py:154]: E211 whitespace before '('

File: es_datastore/elastic_search.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+            (chain.from_iterable([[index_header,
+                                   json.dumps(_generate_multi_entity_es_query(entities=entities,
+                                                                              text=each,
+                                                                              fuzziness_threshold=fuzziness_threshold))]
```
Comment (haptik-deployment) [inline es_datastore/elastic_search.py:158]: E501 line too long (120 > 119 characters)

File: es_datastore/elastic_search.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+        Args:
+            fuzzy_setting (int or str): Can be int or "auto" or "auto:<int>,<int>"
+
+        Returns:
+             int or str: fuzziness as int when ES version < 6.2
+                         otherwise the input is returned as it is
+        """
+        if isinstance(fuzzy_setting, six.string_types):
+            if constants.ELASTICSEARCH_VERSION_MAJOR > 6 or \
+                    (constants.ELASTICSEARCH_VERSION_MAJOR == 6 and
```
Comment (haptik-deployment) [inline es_datastore/elastic_search.py:232]: W504 line break after binary operator

File: es_datastore/queries.py [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+
+def _generate_multi_entity_es_query(entities, text,
+                                    fuzziness_threshold=1,
+                                    language_script=ENGLISH_LANG,
+                                    size=constants.ELASTICSEARCH_SEARCH_SIZE,
+                                    as_json=False):
+    """
+    Generates compound elasticsearch boolean filter search query dictionary
+    for a text for multiple entity_data.
+    The query generated searches for entity_name in the index and returns search results for the matched word (of sentence)
```
Comment (haptik-deployment) [inline es_datastore/queries.py:23]: E501 line too long (123 > 119 characters)

File: es_datastore/queries.py [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+                    e[entity_name]['variant'].extend(
+                        [variant for variant in hit['highlight']['variants']])
+
+                for each_entity in e.keys():
+                    entity_values = e[each_entity]['value']
+                    entity_variants = e[each_entity]['variant']
+                    entity_variants_to_values = collections.OrderedDict()
+
+                    for value, variant in zip(entity_values, entity_variants):
+                        variant = re.sub('\s+', ' ', variant.strip())
```
Comment (haptik-deployment) [inline es_datastore/queries.py:225]: W605 invalid escape sequence '\s'

File: es_datastore/elastic_search.py [hunk 6]
```diff
@@ -0,0 +1,237 @@
+import json
```
Comment (haptik-deployment) [inline es_datastore/elastic_search.py:1]: W1618 [34mimport missing `from __future__ import absolute_import`[0m

File: es_datastore/tests/test_elastic_search.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
+from es_datastore.elastic_search import ElasticSearchDataStore
```
Comment (haptik-deployment) [inline es_datastore/tests/test_elastic_search.py:1]: W1618 [34mimport missing `from __future__ import absolute_import`[0m

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

File: es_datastore/tests/test_queries.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
+from __future__ import absolute_import
+
+import json
+from django.test import TestCase
+
+from es_datastore.queries import _parse_multi_entity_es_results, \
```
Comment (haptik-deployment) [inline es_datastore/tests/test_queries.py:6]: F401 'es_datastore.queries._generate_multi_entity_es_query' imported but unused

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

File: ner_v2/detectors/textual/elastic_search.py [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+        Args:
+            fuzzy_setting (int or str): Can be int or "auto" or "auto:<int>,<int>"
+
+        Returns:
+             int or str: fuzziness as int when ES version < 6.2
+                         otherwise the input is returned as it is
+        """
+        if isinstance(fuzzy_setting, six.string_types):
+            if constants.ELASTICSEARCH_VERSION_MAJOR > 6 or \
+                    (constants.ELASTICSEARCH_VERSION_MAJOR == 6 and
```
Comment (haptik-deployment) [inline ner_v2/detectors/textual/elastic_search.py:234]: W504 line break after binary operator

File: ner_v2/detectors/textual/queries.py [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+                    entity_dict[entity_name]['variant'].extend(
+                        [variant for variant in hit['highlight']['variants']])
+
+                for each_entity in entity_dict.keys():
+                    entity_values = entity_dict[each_entity]['value']
+                    entity_variants = entity_dict[each_entity]['variant']
+                    entity_variants_to_values = collections.OrderedDict()
+
+                    for value, variant in zip(entity_values, entity_variants):
+                        variant = re.sub('\s+', ' ', variant.strip())
```
Comment (haptik-deployment) [inline ner_v2/detectors/textual/queries.py:225]: W605 invalid escape sequence '\s'

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

File: ner_v2/detectors/textual/elastic_search.py [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+            fuzzy_setting (int or str): Can be int or "auto" or "auto:<int>,<int>"
+
+        Returns:
+             int or str: fuzziness as int when ES version < 6.2
+                         otherwise the input is returned as it is
+        """
+        if isinstance(fuzzy_setting, six.string_types):
+            if constants.ELASTICSEARCH_VERSION_MAJOR > 6 \
+                    or (constants.ELASTICSEARCH_VERSION_MAJOR == 6
+                        and
```
Comment (haptik-deployment) [inline ner_v2/detectors/textual/elastic_search.py:235]: W504 line break after binary operator

**Comment by haptik-deployment** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Review by amansrivastava17 — APPROVED** [review]

**Pull request merged**